src/Transport/Topology/ITopology.cs

```diff
@@ -12,9 +16,87 @@ public interface ITopology {
         // TODO: CriticalError no longer passed in to MessagePumpFactory. Ensure that Core is doing pushMessages.OnCriticalError(error);
         Func<IPushMessages> GetMessagePumpFactory();
         Func<IDispatchMessages> GetDispatcherFactory();
+        StartupCheckResult ApplyPreStartupChecks();
         IManageSubscriptions GetSubscriptionManager();
         OutboundRoutingPolicy GetOutboundRoutingPolicy();
         bool HasNativePubSubSupport { get; }
         bool HasSupportForCentralizedPubSub { get;}
     }
+
+    // NOTE: where do we move this components?
+    public interface IStartupCheck
+    {
+        // NOTE: to make awaitable?
+        StartupCheckResult Apply();
+    }
+
+    class CompositeStartupCheck : IStartupCheck
+    {
+        private readonly IEnumerable<IStartupCheck> _checks;
+
+        public CompositeStartupCheck(IEnumerable<IStartupCheck> checks)
+        {
+            _checks = checks;
+        }
+
+        public StartupCheckResult Apply()
+        {
+            foreach (var check in _checks)
+            {
+                var result = check.Apply();
+                if (!result.Succeeded)
+                    return result;
+            }
+
+            return StartupCheckResult.Success;
+        }
+    }
+
+    class NamespacesConfigurationCheck : IStartupCheck
+    {
+        private readonly INamespacePartitioningStrategy namespacePartitioningStrategy;
+        private readonly ReadOnlySettings settings;
+
+        public NamespacesConfigurationCheck(INamespacePartitioningStrategy namespacePartitioningStrategy, ReadOnlySettings settings)
+        {
+            this.namespacePartitioningStrategy = namespacePartitioningStrategy;
+            this.settings = settings;
+        }
+
+        public StartupCheckResult Apply()
+        {
+            var namespaces = settings.Get<List<string>>(WellKnownConfigurationKeys.Topology.Addressing.Partitioning.Namespaces);
+            var endpointName = settings.Get<EndpointName>();
+            
+            var configuredNamespaces = namespacePartitioningStrategy.GetNamespaces(endpointName.ToString(), PartitioningIntent.Creating);
+
+            return namespaces.Count != configuredNamespaces.Count() ?
+                StartupCheckResult.Failed("...") : StartupCheckResult.Success;
```

Comment:
What I'm trying to say is that each class that implement INamespacePartitioningStrategy interface ensures validity checks into its constructor. Ex: [ReplicatedNamespacePartitioningStrategy](https://github.com/Particular/NServiceBus.AzureServiceBus/blob/develop/src/Transport/Addressing/Partitioning/Strategies/ReplicatedNamespacePartitioningStrategy.cs#L14)

If I resolve INamespacePartitioningStrategy implementation with the container, validity checks have already applied, and my StartupCheck is useless. Right? Or am I misunderstending something?